File: uHunt/C++src/easy/SecretResearch.cpp
using namespace std;
#include<cstdio>
#include<cstring>

bool test=true;

int main()
{
    if(test) freopen("../../data/easy/SecretResearch.in","r",stdin);

    char s [2000];
    int casos; scanf("%d",&casos);
    while(casos--)
    {
        scanf("%s",s);
        int length=strlen(s);

        if(strcmp(s,"4")==0 || strcmp(s,"78")==0 || strcmp(s,"1")==0) printf("+");
        else if(s[length-2]=='3' && s[length-1]=='5') printf("-");
        else if(s[0]=='9'&& s[length-1]=='4') printf("*");
        else if(s[0]=='1' && s[1]=='9' && s[2]=='0') printf("?");
        printf("\n");

    }
    return 0;
}

File: uHunt/src/easy/SecretResearch.java
package easy;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class SecretResearch
{
	private static BufferedReader br;
	private static boolean test=true;
	
	public static void main(String[] args) throws IOException
	{	
		if(test) br=new BufferedReader(new FileReader(new File("./data/easy/SecretResearch.in")));
		else br=new BufferedReader(new InputStreamReader(System.in));
		solve();
		br.close();
	}

	private static void solve() throws IOException
	{
		int casos=Integer.parseInt(br.readLine());
		
		while(casos-- != 0)
		{
			String exp=br.readLine();
			if(exp.equals("1") || exp.equals("4") || exp.equals("78")) System.out.println("+");
			else if(exp.endsWith("35")) System.out.println("-");
			else if(exp.startsWith("9") && exp.endsWith("4")) System.out.println("*");
			else if(exp.startsWith("190")) System.out.println("?");
		}
	}
	
}